tests/e2e/common/kubernetes.go

```diff
@@ -164,27 +165,46 @@ func CleanupNode(ctx context.Context, cs clientset.Interface) {
 	}
 }
 
-func CleanupNodeFeatureRules(ctx context.Context, cli *nfdclient.Clientset, namespace string) {
-	// Drop NodeFeatureRule objects
+func CleanupNodeFeatures(ctx context.Context, cli *nfdclient.Clientset, namespace string) {
+	// Clean NodeFeatureRule objects
 	nfrs, err := cli.NfdV1alpha1().NodeFeatureRules().List(ctx, metav1.ListOptions{})
```

Comment:
Given that each of these have multiple levels of nesting, let's move this to a separate `cleanupNodeFeatureRules` function with a quick return if `errors.IsNotFound`?